sourcecode/apis/contentauthor/app/Http/Controllers/Admin/LibraryUpgradeController.php

```diff
@@ -176,4 +179,65 @@ public function deleteLibrary(H5PLibrary $library): Response
 
         return response()->noContent();
     }
+
+    public function checkLibrary(H5PLibrary $library): View
+    {
+        $h5pDataFolderName = $library->getLibraryString(true);
+        $disk = Storage::disk();
+        $libsFolder = $disk->path('libraries');
+        $libFolder = $disk->path('libraries/' . $h5pDataFolderName);
+
+        /** @var H5PValidator $validator */
+        $validator = resolve(H5PValidator::class);
+        $libraryData = [];
+        try {
+            $libraryData = $validator->getLibraryData($h5pDataFolderName, $libFolder, $libsFolder);
```

Comment:
What kind of exception is being caught here? When I look at the code for `H5PValidator::getLibraryData`, I get the impression that any exception thrown from it must result from Laravel's error handling. If that's the case, it should probably be noted in a comment that this is a workaround for programming errors in H5PValidator.

Reply:
The validator does not check that the folder exists before trying to read files in it, this results in an exception. I've updated the code and now checks if the library folder exists before calling the `getLibraryData` function